python/tvm/relay/quantize/quantize.py

```diff
@@ -0,0 +1,258 @@
+from __future__ import absolute_import
+import math
+import numpy as np
+
+from . import _quantize
+from .. import expr as _expr
+from .. import ir_pass as _ir_pass
+from .. import op as _op
+from .. import build_module as _build
+from ..base import register_relay_node
+from ..._ffi.function import register_func
+
+
+class QAnnotateKind(object):
+    """Denote the kind of annotation field, corresponding
+    to different nbit configure."""
+    INPUT = 1
+    WEIGHT = 2
+    ACTIVATION = 3
+
+
+class QConfig(object):
+    """Configuration for quantization.
+
+    Parameters
+    ---------
+    nbit_dict: dict of QAnnotateKind -> int
+        Number of bit for every kind of annotate field.
+
+    global_scale: float
+        The global scale for calibration.
+
+    skip_k_conv: int
+        The number of skipped conv2d.
+    """
+    current = None
+
+    def __init__(self, nbit_dict, global_scale, skip_k_conv):
+        if nbit_dict is None:
+            nbit_dict = {
+                QAnnotateKind.INPUT: 8,
+                QAnnotateKind.WEIGHT: 8,
+                QAnnotateKind.ACTIVATION: 32
+            }
+        self.nbit_dict = nbit_dict
+        self.global_scale = global_scale
+        self.skip_k_conv = skip_k_conv
+        self.counter = 0
+
+    def __enter__(self):
+        self._old_scope = QConfig.current
+        QConfig.current = self
+        return self
+
+    def __exit__(self, ptype, value, trace):
+        assert self._old_scope
+        QConfig.current = self._old_scope
+
+
+QConfig.current = QConfig(None, 8.0, 1)
+
+
+def qconfig(nbit_dict=None, global_scale=8.0, skip_k_conv=1):
+    """Configuration for quantization.
+
+    Parameters
+    ---------
+    nbit_dict: dict of QAnnotateKind -> int
+        Number of bit for every kind of annotate field.
+
+    global_scale: float
+        The global scale for calibration.
+
+    skip_k_conv: int
+        The number of skipped conv2d.
+    """
+    return QConfig(nbit_dict, global_scale, skip_k_conv)
+
+
+def get_current_qconfig():
+    """Get the current quantization configuration."""
+    return QConfig.current
+
+
+@register_func("relay.quantize.get_config_bit")
+def get_config_bit(kind):
```

Comment:
move to _quantize.py

Reply:
I guess it will cause circular reference